azurerm/resource_arm_application_gateway.go

```diff
@@ -1865,6 +1971,14 @@ func flattenApplicationGatewayRequestRoutingRules(input *[]network.ApplicationGa
 	for _, config := range *input {
 		if props := config.ApplicationGatewayRequestRoutingRulePropertiesFormat; props != nil {
 
+			if props.BackendAddressPool != nil && props.RedirectConfiguration != nil {
```

Comment:
since this is a pointer to a string (`*string`) - I think we also want to check these aren't `*props.BackendAddressPool != ""` too, in an inner-check

Reply:
Hi @tombuildsstuff,  I total get the string example - I need to get my C++ head back on ;). However, in the example, `props.BackendAddressPool` here is a `SubResource *`.  So should I also be doing `props.BackendAddressPool != nil' and `props.BackendAddressPool.ID != nil`  and `*props.BackendAddressPool.ID != nil`  ? Or am I losing the plot?

Reply:
ah I'd missed this was a `*SubResource` in which case, yes we'll want to make this:

```
hasBackendAddressPool := props.BackendAddressPool != nil && props.BackendAddressPool.ID != nil && *props.BackendAddressPool.ID != ""
hasRedirectConfiguration := props.RedirectConfiguration != nil && props.RedirectConfiguration.ID != nil && * props.RedirectConfiguration.ID != ""
if hasBackendAddressPool && hasRedirectConfiguration {
  # ...
}
```

Reply:
Done, I think